fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java

```diff
@@ -156,121 +156,294 @@ public ColumnStatistic visitCall(CallOperator call, Void context) {
                                     left.getMinValue() * right.getMaxValue()),
                             left.getMaxValue() * right.getMinValue()), left.getMaxValue() * right.getMaxValue());
-                    return new ColumnStatistic(multiplyMinValue, multiplyMaxValue, nullsFraction,
-                            callOperator.getType().getTypeSize(), distinctValues);
+                    break;
                 case FunctionSet.DIVIDE:
-                    double divideMinValue = Math.min(Math.min(
-                            Math.min(left.getMinValue() / divisorNotZero(right.getMinValue()),
-                                    left.getMinValue() / divisorNotZero(right.getMaxValue())),
-                            left.getMaxValue() / divisorNotZero(right.getMinValue())),
+                    minValue = Math.min(Math.min(
+                                    Math.min(left.getMinValue() / divisorNotZero(right.getMinValue()),
+                                            left.getMinValue() / divisorNotZero(right.getMaxValue())),
+                                    left.getMaxValue() / divisorNotZero(right.getMinValue())),
                             left.getMaxValue() / divisorNotZero(right.getMaxValue()));
-                    double divideMaxValue = Math.max(Math.max(
-                            Math.max(left.getMinValue() / divisorNotZero(right.getMinValue()),
-                                    left.getMinValue() / divisorNotZero(right.getMaxValue())),
-                            left.getMaxValue() / divisorNotZero(right.getMinValue())),
+                    maxValue = Math.max(Math.max(
+                                    Math.max(left.getMinValue() / divisorNotZero(right.getMinValue()),
+                                            left.getMinValue() / divisorNotZero(right.getMaxValue())),
+                                    left.getMaxValue() / divisorNotZero(right.getMinValue())),
                             left.getMaxValue() / divisorNotZero(right.getMaxValue()));
-                    return new ColumnStatistic(divideMinValue, divideMaxValue, nullsFraction,
-                            callOperator.getType().getTypeSize(),
-                            distinctValues);
-                // use child column statistics for now
-                case FunctionSet.SUBSTRING:
-                    return left;
+                    break;
+                case FunctionSet.MOD:
+                case FunctionSet.FMOD:
+                case FunctionSet.PMOD:
+                    minValue = -Math.max(Math.abs(right.getMinValue()), Math.abs(right.getMaxValue()));
+                    maxValue = -minValue;
+                    break;
                 default:
                     return ColumnStatistic.unknown();
             }
+
+            return new ColumnStatistic(minValue, maxValue, nullsFraction, averageRowSize, distinctValues);
         }
 
         private ColumnStatistic multiaryExpressionCalculate(CallOperator callOperator,
```

Comment:
support if_null, case_when?

Reply:
add `ifnull` and `nullif`
case_when is already supported